plugins/polio/tasks/api/refresh_lqas_data.py

```diff
@@ -0,0 +1,218 @@
+            task.progress_value = validated_data["progress_value"]
+        if has_end_value:
+            task.end_value = validated_data["end_value"]
+        task.save()
+        return task
+
+
+class CustomTaskSearchFilterBackend(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        search = request.query_params.get("search")
+        if search:
+            query = (
+                Q(name__icontains=search)
+                | Q(status__icontains=search)
+                | Q(launcher__first_name__icontains=search)
+                | Q(launcher__username__icontains=search)
+                | Q(launcher__last_name__icontains=search)
+            )
+            return queryset.filter(query)
+
+        return queryset
+
+
+class RefreshLQASDataViewset(ModelViewSet):
+    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.POLIO, permission.POLIO_CONFIG)]  # type: ignore
+    http_method_names = ["get", "post", "patch"]
+    model = Task
+    filter_backends = [
+        filters.OrderingFilter,
+        CustomTaskSearchFilterBackend,
+        DjangoFilterBackend,
+    ]
+    filterset_fields = {"created_at": ["gte"], "ended_at": ["exact"], "started_at": ["gte"], "status": ["exact"]}
+    ordering_fields = ["created_at", "ended_at", "name", "started_at", "status"]
+    ordering = ["-started_at"]
+
+    def get_serializer_class(self):
+        if self.request.method == "POST":
+            return RefreshLQASDataSerializer
+        return ExternalTaskSerializer
+
+    def get_queryset(self):
+        return Task.objects.all()
+
+    def create(self, request):
+        serializer = RefreshLQASDataSerializer(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        user = request.user
+        data = serializer.validated_data
+        started_at = datetime.now()
+        country_id = data.get("country_id", None)
+        name = f"{TASK_NAME}-{country_id}" if country_id is not None else TASK_NAME
+        task = Task.objects.create(
+            launcher=user,
+            account=user.iaso_profile.account,
+            name=name,
+            status=RUNNING,
+            external=True,
+            started_at=started_at,
+            should_be_killed=False,
+        )
+        status = self.refresh_lqas_data(country_id, task.id)
+        task.status = status
+        task.save()
+        return Response({"task": TaskSerializer(instance=task).data})
+
+    def refresh_lqas_data(self, country_id=None, task_id=None):
+        transport = RequestsHTTPTransport(
+            url=settings.OPENHEXA_URL,
+            verify=True,
+            headers={"Authorization": f"Bearer {settings.OPENHEXA_TOKEN}"},
+        )
+        client = Client(transport=transport, fetch_schema_from_transport=True)
+        get_runs = gql(
+            """
+        query pipeline {
+            pipeline(id: "e5b9a3bb-a22c-461f-b000-4dfa8d1cc445"){
```

Comment:
Does the pipeline `uuid` needs to be hardcoded there ?

Reply:
At least put it in a constant somewhere or better in a settings var (or the env)

Reply:
Good catch, it's already in `settings.LQAS_PIPELINE`, I missed this occurence it when I refactored